p2p/starknet/client.go

```diff
@@ -14,6 +14,8 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
+const unmarshalMaxSize = 15 * utils.Megabyte
```

Comment:
Wondering why the limit is set to 15mb? Are we sure there won't be messages bigger than that?

Reply:
Idk, max message that we saw is 12 Mb, I set it +3 Mb just to be aware of next time when it happens

Reply:
Are there a downsides to set it higher? 

I feel we will be hit by this again 🤔 
Should [p2p spec](https://github.com/starknet-io/starknet-p2p-specs) define max message size?
Maybe we should check & set it at the similar level as Ethereum client?

Reply:
> Should [p2p spec](https://github.com/starknet-io/starknet-p2p-specs) define max message size?

That's a good point, initially we discussed it as 2Mb (because there were rumors about max possible size that's not controlled by a caller) but then decided to get rid from the spec. 

I think we should do the following:
- implement recent changes from the spec (it will reduce size of messages)
- come up with some max message size based on starknet constraints (max number of tx-s per block etc.) and what we get from real-world tests